### PR TITLE
Add listmodel functions

### PR DESCRIPTION
**Pull request by midwan**
Added two virtual functions in listmodel:
- add
- clear_elements

I think these makes sense to have as a minimum model, since they are almost always needed.

### CODE REVIEW

File: include/guisan/listmodel.hpp [hunk 1]
```diff
@@ -92,6 +92,12 @@ namespace gcn
          * @return An element as a string at the a certain index.
          */
         virtual std::string getElementAt(int i) = 0;
+
+        // Add a new element
+        virtual void add(const std::string& str) {}
+
+        // Clear all elements
+        virtual void clear_elements() {}
```
Comment (Jarod42) [inline include/guisan/listmodel.hpp:100]: Better to have `= 0;` IMO for both.
but maybe inherit from this class would be better.

Comment (midwan) [reply include/guisan/listmodel.hpp]: Adapter the code accordingly. Also added = default for the destructor

**Comment by Jarod42** [comment]
From interface point of view, I disagree with the usage which requires dummy implementation (Broken Liskov's principle).

I have just created https://github.com/kallisti5/guisan/pull/132 for stringlistmodel (which is indeed a useful class permitting to mutate the list).

I think that PR might be closed.

**Comment by midwan** [comment]
OK, let's go with stringlistmodel (I did the same in my project as well)

**Pull request closed without merging**